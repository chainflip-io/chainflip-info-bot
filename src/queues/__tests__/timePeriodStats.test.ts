import { utc } from '@date-fns/utc';
import { BigNumber } from 'bignumber.js';
import { Job } from 'bullmq';
import { endOfDay } from 'date-fns';
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
import getSwapVolumeStats from '../../queries/swapVolume.js';
import { config } from '../timePeriodStats.js';

vi.mock('../../queries/swapVolume.js');

describe('time period stats', () => {
  beforeEach(() => {
    // tell vitest we use mocked time
    vi.useFakeTimers();
  });

  afterEach(() => {
    // restoring date after each test run
    vi.useRealTimers();
  });

  describe('initialize', () => {
    it('primes the queue', async () => {
      vi.setSystemTime(new Date('2024-10-25T12:34:56Z'));

      const queue = { add: vi.fn() };

      await config.initialize?.(queue as any);
      expect(queue.add.mock.lastCall).toMatchInlineSnapshot(`
        [
          "timePeriodStats",
          {
            "endOfPeriod": 1729900799999,
            "sendWeeklySummary": false,
          },
          {
            "delay": 41103999,
            "jobId": "timePeriodStatsSingleton-1729900799999",
          },
        ]
      `);
    });

    it('primes the queue with the weekly job', async () => {
      vi.setSystemTime(new Date('2024-10-27T12:34:56Z'));

      const queue = { add: vi.fn() };

      await config.initialize?.(queue as any);
      expect(queue.add.mock.lastCall).toMatchInlineSnapshot(`
        [
          "timePeriodStats",
          {
            "endOfPeriod": 1730073599999,
            "sendWeeklySummary": true,
          },
          {
            "delay": 41103999,
            "jobId": "timePeriodStatsSingleton-1730073599999",
          },
        ]
      `);
    });
  });

  describe('processJob', () => {
    it('processes a daily job', async () => {
      vi.setSystemTime(new Date('2024-10-25T12:34:56Z'));
      vi.mocked(getSwapVolumeStats)
        .mockResolvedValueOnce({
          flipBurned: new BigNumber('1000.00'),
          lpFees: new BigNumber('2000.00'),
          networkFees: new BigNumber('3000.00'),
          swapVolume: new BigNumber('4000.00'),
        })
        .mockRejectedValue(Error('unexpected call'));

      const dispatchJobs = vi.fn();

      const endOfPeriod = endOfDay(new Date('2024-10-25T12:34:56Z'), { in: utc });

      await config.processJob(dispatchJobs)({
        data: { endOfPeriod: endOfPeriod.valueOf(), sendWeeklySummary: false },
      } as Job<JobData['timePeriodStats'], any, any>);

      expect(dispatchJobs.mock.calls).toMatchSnapshot();
      expect(vi.mocked(getSwapVolumeStats).mock.calls).toMatchInlineSnapshot(`
        [
          [
            2024-10-25T00:00:00.000Z,
            2024-10-25T23:59:59.999Z,
          ],
        ]
      `);
    });

    it('processes a daily job without a flip burn', async () => {
      vi.setSystemTime(new Date('2024-10-25T12:34:56Z'));
      vi.mocked(getSwapVolumeStats)
        .mockResolvedValueOnce({
          flipBurned: null,
          lpFees: new BigNumber('2000.00'),
          networkFees: new BigNumber('3000.00'),
          swapVolume: new BigNumber('4000.00'),
        })
        .mockRejectedValue(Error('unexpected call'));

      const dispatchJobs = vi.fn();

      const endOfPeriod = endOfDay(new Date('2024-10-25T12:34:56Z'), { in: utc });

      await config.processJob(dispatchJobs)({
        data: { endOfPeriod: endOfPeriod.valueOf(), sendWeeklySummary: false },
      } as Job<JobData['timePeriodStats'], any, any>);

      expect(dispatchJobs.mock.calls).toMatchSnapshot();
    });

    it('processes a daily and weekly job', async () => {
      vi.setSystemTime(new Date('2024-10-27T12:34:56Z'));
      vi.mocked(getSwapVolumeStats)
        .mockResolvedValueOnce({
          flipBurned: new BigNumber('1000.00'),
          lpFees: new BigNumber('2000.00'),
          networkFees: new BigNumber('3000.00'),
          swapVolume: new BigNumber('4000.00'),
        })
        .mockResolvedValueOnce({
          flipBurned: new BigNumber('1000.00').times(7),
          lpFees: new BigNumber('2000.00').times(7),
          networkFees: new BigNumber('3000.00').times(7),
          swapVolume: new BigNumber('4000.00').times(7),
        })
        .mockRejectedValue(Error('unexpected call'));

      const dispatchJobs = vi.fn();

      const endOfPeriod = endOfDay(new Date('2024-10-27T12:34:56Z'), { in: utc });

      await config.processJob(dispatchJobs)({
        data: { endOfPeriod: endOfPeriod.valueOf(), sendWeeklySummary: true },
      } as Job<JobData['timePeriodStats'], any, any>);

      expect(dispatchJobs.mock.calls).toMatchSnapshot();
      expect(vi.mocked(getSwapVolumeStats).mock.calls).toMatchInlineSnapshot(`
        [
          [
            2024-10-27T00:00:00.000Z,
            2024-10-27T23:59:59.999Z,
          ],
          [
            2024-10-21T00:00:00.000Z,
            2024-10-27T23:59:59.999Z,
          ],
        ]
      `);
    });
  });
});
