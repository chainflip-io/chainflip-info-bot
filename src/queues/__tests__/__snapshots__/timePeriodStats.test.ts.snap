// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`time period stats > processJob > processes a daily and weekly job 1`] = `
[
  [
    [
      {
        "data": [
          {
            "data": {
              "endOfPeriod": 1730073599999,
              "sendWeeklySummary": true,
            },
            "name": "timePeriodStats",
            "opts": {
              "deduplication": {
                "id": "timePeriodStats-1730073599999",
              },
              "delay": 41103999,
            },
          },
        ],
        "name": "scheduler",
      },
      {
        "data": {
          "message": "🗓️ On <strong>2024-10-27</strong>, we had:
📊 <strong>$4,000.00</strong> in total volume
🌐 <strong>$3,000.00</strong> of network fees
🤑 <strong>$2,000.00</strong> of LP fees
⚡️ <strong>$5,000.00</strong> of boost fees
🔥 <strong>1000.00</strong> FLIP burned",
          "platform": "telegram",
          "validationData": {
            "name": "DAILY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "🗓️ On **2024-10-27**, we had:
📊 **$4,000.00** in total volume
🌐 **$3,000.00** of network fees
🤑 **$2,000.00** of LP fees
⚡️ **$5,000.00** of boost fees
🔥 **1000.00** FLIP burned",
          "platform": "discord",
          "validationData": {
            "name": "DAILY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "🗓️ On 2024-10-27, we had:
📊 $4,000.00 in total volume
🌐 $3,000.00 of network fees
🤑 $2,000.00 of LP fees
⚡️ $5,000.00 of boost fees
🔥 1000.00 FLIP burned",
          "platform": "twitter",
          "validationData": {
            "name": "DAILY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for 2024-10-27 are in:
🥇 $1,000.00 <a href="https://scan.chainflip.io/lps/cFMboYsd4HvERKXX11LyvZXuTcQzV7KAe9ipP4La5vUs8fd4e"><strong>ChainflipGod</strong></a> (10.00%)
🥈 $1,000.00 <a href="https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh"><strong>cFNz…MMVh</strong></a> (10.00%)
🥉 $1,000.00 <a href="https://scan.chainflip.io/lps/cFKjURUE4jdxHgcKb4uBnKiY9Pkx2yuvQuRVfTDFh5j5eUgyN"><strong>cFKj…UgyN</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh"><strong>cFNz…MMVh</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFJXT4WEEdfiShje4z9JMwAvMiMTu7nioPgXsE9o1KqdVrzLg"><strong>cFJX…rzLg</strong></a> (10.00%)
",
          "platform": "telegram",
          "validationData": {
            "name": "DAILY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for 2024-10-27 are in:
🥇 $1,000.00 [**ChainflipGod**](https://scan.chainflip.io/lps/cFMboYsd4HvERKXX11LyvZXuTcQzV7KAe9ipP4La5vUs8fd4e) (10.00%)
🥈 $1,000.00 [**cFNz…MMVh**](https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh) (10.00%)
🥉 $1,000.00 [**cFKj…UgyN**](https://scan.chainflip.io/lps/cFKjURUE4jdxHgcKb4uBnKiY9Pkx2yuvQuRVfTDFh5j5eUgyN) (10.00%)
🏅 $1,000.00 [**cFNz…MMVh**](https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh) (10.00%)
🏅 $1,000.00 [**cFJX…rzLg**](https://scan.chainflip.io/lps/cFJXT4WEEdfiShje4z9JMwAvMiMTu7nioPgXsE9o1KqdVrzLg) (10.00%)
",
          "platform": "discord",
          "validationData": {
            "name": "DAILY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for 2024-10-27 are in:
🥇 $1,000.00 ChainflipGod (10.00%)
🥈 $1,000.00 cFNz…MMVh (10.00%)
🥉 $1,000.00 cFKj…UgyN (10.00%)
🏅 $1,000.00 cFNz…MMVh (10.00%)
🏅 $1,000.00 cFJX…rzLg (10.00%)
",
          "platform": "twitter",
          "validationData": {
            "name": "DAILY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "🗓️ For the week ending <strong>2024-10-27</strong>, we had:
📊 <strong>$28,000.00</strong> in total volume
🌐 <strong>$21,000.00</strong> of network fees
🤑 <strong>$14,000.00</strong> of LP fees
⚡️ <strong>$5,000.00</strong> of boost fees
🔥 <strong>7000.00</strong> FLIP burned",
          "platform": "telegram",
          "validationData": {
            "name": "WEEKLY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "🗓️ For the week ending **2024-10-27**, we had:
📊 **$28,000.00** in total volume
🌐 **$21,000.00** of network fees
🤑 **$14,000.00** of LP fees
⚡️ **$5,000.00** of boost fees
🔥 **7000.00** FLIP burned",
          "platform": "discord",
          "validationData": {
            "name": "WEEKLY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "🗓️ For the week ending 2024-10-27, we had:
📊 $28,000.00 in total volume
🌐 $21,000.00 of network fees
🤑 $14,000.00 of LP fees
⚡️ $5,000.00 of boost fees
🔥 7000.00 FLIP burned",
          "platform": "twitter",
          "validationData": {
            "name": "WEEKLY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for the week are in:
🥇 $1,000.00 <a href="https://scan.chainflip.io/lps/cFMboYsd4HvERKXX11LyvZXuTcQzV7KAe9ipP4La5vUs8fd4e"><strong>ChainflipGod</strong></a> (10.00%)
🥈 $1,000.00 <a href="https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh"><strong>cFNz…MMVh</strong></a> (10.00%)
🥉 $1,000.00 <a href="https://scan.chainflip.io/lps/cFKjURUE4jdxHgcKb4uBnKiY9Pkx2yuvQuRVfTDFh5j5eUgyN"><strong>cFKj…UgyN</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh"><strong>cFNz…MMVh</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFJXT4WEEdfiShje4z9JMwAvMiMTu7nioPgXsE9o1KqdVrzLg"><strong>cFJX…rzLg</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFMxkJ79ka5Bu8PTxwMm9E2U8bKR5fwaPuNQBErXoaYvFYwvB"><strong>cFMx…YwvB</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFMBE4zhwZ9PQ7xBupusy2mLerkSBhU9oAxoMiCpw3kKAR4Kz"><strong>cFMB…R4Kz</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFLEFzKb4QCruqMn5xrCm2sPFt2taXWrvuSDTvuq6U3etkcPP"><strong>cFLE…kcPP</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFLGvPhhrribWCx9id5kLVqwiFK4QiVNjQ6ViyaRFF2Nrgq7j"><strong>cFLG…gq7j</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFLT2BLiSijog9mvFipfPapisweJ8MMMTQisCiNKAEsBQnn5t"><strong>cFLT…nn5t</strong></a> (10.00%)
",
          "platform": "telegram",
          "validationData": {
            "name": "WEEKLY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for the week are in:
🥇 $1,000.00 [**ChainflipGod**](https://scan.chainflip.io/lps/cFMboYsd4HvERKXX11LyvZXuTcQzV7KAe9ipP4La5vUs8fd4e) (10.00%)
🥈 $1,000.00 [**cFNz…MMVh**](https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh) (10.00%)
🥉 $1,000.00 [**cFKj…UgyN**](https://scan.chainflip.io/lps/cFKjURUE4jdxHgcKb4uBnKiY9Pkx2yuvQuRVfTDFh5j5eUgyN) (10.00%)
🏅 $1,000.00 [**cFNz…MMVh**](https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh) (10.00%)
🏅 $1,000.00 [**cFJX…rzLg**](https://scan.chainflip.io/lps/cFJXT4WEEdfiShje4z9JMwAvMiMTu7nioPgXsE9o1KqdVrzLg) (10.00%)
🏅 $1,000.00 [**cFMx…YwvB**](https://scan.chainflip.io/lps/cFMxkJ79ka5Bu8PTxwMm9E2U8bKR5fwaPuNQBErXoaYvFYwvB) (10.00%)
🏅 $1,000.00 [**cFMB…R4Kz**](https://scan.chainflip.io/lps/cFMBE4zhwZ9PQ7xBupusy2mLerkSBhU9oAxoMiCpw3kKAR4Kz) (10.00%)
🏅 $1,000.00 [**cFLE…kcPP**](https://scan.chainflip.io/lps/cFLEFzKb4QCruqMn5xrCm2sPFt2taXWrvuSDTvuq6U3etkcPP) (10.00%)
🏅 $1,000.00 [**cFLG…gq7j**](https://scan.chainflip.io/lps/cFLGvPhhrribWCx9id5kLVqwiFK4QiVNjQ6ViyaRFF2Nrgq7j) (10.00%)
🏅 $1,000.00 [**cFLT…nn5t**](https://scan.chainflip.io/lps/cFLT2BLiSijog9mvFipfPapisweJ8MMMTQisCiNKAEsBQnn5t) (10.00%)
",
          "platform": "discord",
          "validationData": {
            "name": "WEEKLY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for the week are in:
🥇 $1,000.00 ChainflipGod (10.00%)
🥈 $1,000.00 cFNz…MMVh (10.00%)
🥉 $1,000.00 cFKj…UgyN (10.00%)
🏅 $1,000.00 cFNz…MMVh (10.00%)
🏅 $1,000.00 cFJX…rzLg (10.00%)
🏅 $1,000.00 cFMx…YwvB (10.00%)
🏅 $1,000.00 cFMB…R4Kz (10.00%)
🏅 $1,000.00 cFLE…kcPP (10.00%)
🏅 $1,000.00 cFLG…gq7j (10.00%)
🏅 $1,000.00 cFLT…nn5t (10.00%)
",
          "platform": "twitter",
          "validationData": {
            "name": "WEEKLY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
    ],
  ],
]
`;

exports[`time period stats > processJob > processes a daily job 1`] = `
[
  [
    [
      {
        "data": [
          {
            "data": {
              "endOfPeriod": 1729900799999,
              "sendWeeklySummary": false,
            },
            "name": "timePeriodStats",
            "opts": {
              "deduplication": {
                "id": "timePeriodStats-1729900799999",
              },
              "delay": 41103999,
            },
          },
        ],
        "name": "scheduler",
      },
      {
        "data": {
          "message": "🗓️ On <strong>2024-10-25</strong>, we had:
📊 <strong>$4,000.00</strong> in total volume
🌐 <strong>$3,000.00</strong> of network fees
🤑 <strong>$2,000.00</strong> of LP fees
⚡️ <strong>$5,000.00</strong> of boost fees
🔥 <strong>1000.00</strong> FLIP burned",
          "platform": "telegram",
          "validationData": {
            "name": "DAILY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "🗓️ On **2024-10-25**, we had:
📊 **$4,000.00** in total volume
🌐 **$3,000.00** of network fees
🤑 **$2,000.00** of LP fees
⚡️ **$5,000.00** of boost fees
🔥 **1000.00** FLIP burned",
          "platform": "discord",
          "validationData": {
            "name": "DAILY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "🗓️ On 2024-10-25, we had:
📊 $4,000.00 in total volume
🌐 $3,000.00 of network fees
🤑 $2,000.00 of LP fees
⚡️ $5,000.00 of boost fees
🔥 1000.00 FLIP burned",
          "platform": "twitter",
          "validationData": {
            "name": "DAILY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for 2024-10-25 are in:
🥇 $1,000.00 <a href="https://scan.chainflip.io/lps/cFMboYsd4HvERKXX11LyvZXuTcQzV7KAe9ipP4La5vUs8fd4e"><strong>ChainflipGod</strong></a> (10.00%)
🥈 $1,000.00 <a href="https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh"><strong>cFNz…MMVh</strong></a> (10.00%)
🥉 $1,000.00 <a href="https://scan.chainflip.io/lps/cFKjURUE4jdxHgcKb4uBnKiY9Pkx2yuvQuRVfTDFh5j5eUgyN"><strong>cFKj…UgyN</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh"><strong>cFNz…MMVh</strong></a> (10.00%)
🏅 $1,000.00 <a href="https://scan.chainflip.io/lps/cFJXT4WEEdfiShje4z9JMwAvMiMTu7nioPgXsE9o1KqdVrzLg"><strong>cFJX…rzLg</strong></a> (10.00%)
",
          "platform": "telegram",
          "validationData": {
            "name": "DAILY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for 2024-10-25 are in:
🥇 $1,000.00 [**ChainflipGod**](https://scan.chainflip.io/lps/cFMboYsd4HvERKXX11LyvZXuTcQzV7KAe9ipP4La5vUs8fd4e) (10.00%)
🥈 $1,000.00 [**cFNz…MMVh**](https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh) (10.00%)
🥉 $1,000.00 [**cFKj…UgyN**](https://scan.chainflip.io/lps/cFKjURUE4jdxHgcKb4uBnKiY9Pkx2yuvQuRVfTDFh5j5eUgyN) (10.00%)
🏅 $1,000.00 [**cFNz…MMVh**](https://scan.chainflip.io/lps/cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh) (10.00%)
🏅 $1,000.00 [**cFJX…rzLg**](https://scan.chainflip.io/lps/cFJXT4WEEdfiShje4z9JMwAvMiMTu7nioPgXsE9o1KqdVrzLg) (10.00%)
",
          "platform": "discord",
          "validationData": {
            "name": "DAILY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for 2024-10-25 are in:
🥇 $1,000.00 ChainflipGod (10.00%)
🥈 $1,000.00 cFNz…MMVh (10.00%)
🥉 $1,000.00 cFKj…UgyN (10.00%)
🏅 $1,000.00 cFNz…MMVh (10.00%)
🏅 $1,000.00 cFJX…rzLg (10.00%)
",
          "platform": "twitter",
          "validationData": {
            "name": "DAILY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
    ],
  ],
]
`;

exports[`time period stats > processJob > processes a daily job without a flip burn or boost fees 1`] = `
[
  [
    [
      {
        "data": [
          {
            "data": {
              "endOfPeriod": 1729900799999,
              "sendWeeklySummary": false,
            },
            "name": "timePeriodStats",
            "opts": {
              "deduplication": {
                "id": "timePeriodStats-1729900799999",
              },
              "delay": 41103999,
            },
          },
        ],
        "name": "scheduler",
      },
      {
        "data": {
          "message": "🗓️ On <strong>2024-10-25</strong>, we had:
📊 <strong>$4,000.00</strong> in total volume
🌐 <strong>$3,000.00</strong> of network fees
🤑 <strong>$2,000.00</strong> of LP fees",
          "platform": "telegram",
          "validationData": {
            "name": "DAILY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "🗓️ On **2024-10-25**, we had:
📊 **$4,000.00** in total volume
🌐 **$3,000.00** of network fees
🤑 **$2,000.00** of LP fees",
          "platform": "discord",
          "validationData": {
            "name": "DAILY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "🗓️ On 2024-10-25, we had:
📊 $4,000.00 in total volume
🌐 $3,000.00 of network fees
🤑 $2,000.00 of LP fees",
          "platform": "twitter",
          "validationData": {
            "name": "DAILY_SWAP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for 2024-10-25 are in:
",
          "platform": "telegram",
          "validationData": {
            "name": "DAILY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for 2024-10-25 are in:
",
          "platform": "discord",
          "validationData": {
            "name": "DAILY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
      {
        "data": {
          "message": "💼 Top LPs for 2024-10-25 are in:
",
          "platform": "twitter",
          "validationData": {
            "name": "DAILY_LP_SUMMARY",
          },
        },
        "name": "messageRouter",
      },
    ],
  ],
]
`;
